app/scripts/controllers/buildConfig.js

```diff
@@ -66,6 +73,32 @@ angular.module('openshiftConsole')
               });
             }
 
+            copyBuildConfigAndEnsureEnv(buildConfig);
+
+            $scope.saveEnvVars = function() {
+              $scope.envVars = _.filter($scope.envVars, 'name');
+              buildStrategy($scope.updatedBuildConfig).env = $scope.envVars;
+
+              DataService
+                .update("buildconfigs", $routeParams.buildconfig, angular.copy($scope.updatedBuildConfig), context)
```

Comment:
Sorry, just noticed: do we need to copy again here? `updatedBuildConfig` is already a copy, right?